operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSource holds cluster-wide information about how to handle registry mirror rules.  The canonical name is `cluster`
+type ImageContentSource struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec ImageContentSourceSpec `json:"spec"`
+}
+
+type ImageContentSourceSpec struct {
+	// MirrorReposForDigests holds the registry/repo mirroring rules that are written to
```

Comment:
You should not be describing the exact implementation - you should be describing the intent of the action.  For instance:

```
<field> allows images referenced by image digest in pods to be retrieved from alternative repository locations.  The image pull specification provided to the pod will be compared the source locations described in <field>, and the image may be pulled from any repository in that list.  The cluster may prefer to pull from the first repository in the list instead of the specified repository, which allows administrators to choose a potentially faster mirror.  Only image pull specifications that contain an image digest will have this behavior applied - tags will continue to be pulled from the specified repository.
```

Probably also need to describe what happens with authentication.

Reply:
Fixed